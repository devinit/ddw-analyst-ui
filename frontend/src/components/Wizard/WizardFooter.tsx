import React, { FunctionComponent } from 'react';
import { Card } from 'react-bootstrap';

type ComponentProps = {
  children?: React.ReactNode;
};
const WizardFooter: FunctionComponent<ComponentProps> = ({ children }) => {
  return <Card.Footer>{children}</Card.Footer>;
};

export { WizardFooter };
