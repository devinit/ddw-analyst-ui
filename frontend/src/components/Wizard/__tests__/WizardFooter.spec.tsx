/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { WizardFooter } from '../WizardFooter';

describe('WizardFooter', () => {
  test('renders the default structure correctly', () => {
    const renderer = TestRenderer.create(<WizardFooter />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with children', () => {
    const renderer = TestRenderer.create(
      <WizardFooter>
        <div>My Child</div>
      </WizardFooter>,
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
