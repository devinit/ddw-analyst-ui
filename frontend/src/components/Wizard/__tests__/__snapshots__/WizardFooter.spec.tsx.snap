// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardFooter renders the default structure correctly 1`] = `
<div
  className="card-footer"
/>
`;

exports[`WizardFooter renders with children 1`] = `
<div
  className="card-footer"
>
  <div>
    My Child
  </div>
</div>
`;
